### PR TITLE
Clean log messages

### PR DESCRIPTION
**Pull request by aserdean**
This PR is intended to clean the wnbd log messages.

- Remove leading `: ` from log messages.

- Fix format specifiers.

- Demote some log messages.

- Print NTSTATUS as hex instead of integer format.

- Remove extraneous arguments passed to log messages

### CODE REVIEW

File: driver/driver.c [hunk 1]
```diff
@@ -89,7 +89,7 @@ DriverEntry(PDRIVER_OBJECT DriverObject,
                                  (PHW_INITIALIZATION_DATA)&WnbdInitData,
                                  NULL);
     if (!NT_SUCCESS(Status)) {
-        WNBD_LOG_ERROR("DriverEntry failure in call to StorPortInitialize. Status: 0x%x", Status);
+        WNBD_LOG_ERROR("DriverEntry failure in call to StorPortInitialize. Status: %x", Status);
```
Comment (petrutlucian94) [inline driver/driver.c:92]: Let's not use "%x", we can stick to "0x%x" or "%#x". Without the "0x" prefix, it's hard to tell if that's a decimal or hexadecimal number if we get something like "Status: 10".

File: driver/driver.c [hunk 2]
```diff
@@ -29,7 +29,7 @@ NTSTATUS
 DriverEntry(PDRIVER_OBJECT DriverObject,
             PUNICODE_STRING RegistryPath)
 {
-    WNBD_LOG_LOUD(": Enter");
+    WNBD_LOG_LOUD("Enter");
```
Comment (petrutlucian94) [inline driver/driver.c:32]: I'm wondering if we should keep those ``` WNBD_LOG_LOUD("Enter");```

File: driver/driver.c [hunk 3]
```diff
@@ -103,8 +103,7 @@ DriverEntry(PDRIVER_OBJECT DriverObject,
     DriverObject->MajorFunction[IRP_MJ_PNP] = 0 != StorPortDispatchPnp ? WnbdDispatchPnp : 0;
     GlobalExt = NULL;
 
-    WNBD_LOG_LOUD(": Exit");
-
+    WNBD_LOG_LOUD("Exit");
```
Comment (petrutlucian94) [inline driver/driver.c:106]: Using one commit per file is weird and pollutes the git log, let's squash them.

File: driver/nbd_dispatch.c [hunk 4]
```diff
@@ -234,7 +240,7 @@ NbdProcessDeviceThreadReplies(_In_ PWNBD_DISK_DEVICE Device)
     }
     KeReleaseSpinLock(&Device->SubmittedReqListLock, Irql);
     if(!Element) {
-        WNBD_LOG_ERROR("Received reply with no matching request tag: 0x%llx",
+        WNBD_LOG_WARN("Received reply with no matching request tag: 0x%llx",
```
Comment (petrutlucian94) [inline driver/nbd_dispatch.c:243]: Let's use a debug message here. This will be printed for every abandoned request and there can be lots of them.

File: driver/nbd_protocol.c [hunk 5]
```diff
@@ -187,16 +189,16 @@ NbdSendOptExportName(_In_ INT Fd,
...
 
     NTSTATUS error = STATUS_SUCCESS;
 
     NbdSendRequest(Fd, NBD_OPT_EXPORT_NAME, strlen(Name), Name);
     CHAR Buf[sizeof(*Flags) + sizeof(*Size)];
     RtlZeroMemory(Buf, sizeof(*Flags) + sizeof(*Size));
     if (NbdReadExact(Fd, Buf, sizeof(Buf), &error) < 0 && Go) {
-        WNBD_LOG_ERROR("Server does not support NBD_OPT_GO and"
-            "dropped connection after sending NBD_OPT_EXPORT_NAME.");
+        WNBD_LOG_INFO("Server does not support NBD_OPT_GO and"
```
Comment (petrutlucian94) [inline driver/nbd_protocol.c:200]: This should be a warning, at least.

File: driver/nbd_protocol.c [hunk 6]
```diff
@@ -228,14 +230,14 @@ NbdNegotiate(_In_ INT* Pfd,
     RtlZeroMemory(Buf, 8);
     NbdReadExact(Fd, Buf, 8, &status);
     if (strcmp(Buf, INIT_PASSWD)) {
-        WNBD_LOG_ERROR("INIT_PASSWD");
+        WNBD_LOG_INFO("INIT_PASSWD");
```
Comment (petrutlucian94) [inline driver/nbd_protocol.c:233]: Let's also update the message, maybe something like "Received NBD INIT_PASSWD". Also, I guess it can be a debug message.

File: driver/nbd_protocol.c [hunk 7]
```diff
@@ -139,16 +140,17 @@ NbdReadHandshakeReply(_In_ INT Fd)
     Retval->Datasize = RtlUlongByteSwap(Retval->Datasize);
 
     if (REPLY_MAGIC != Retval->Magic) {
-        WNBD_LOG_ERROR("Received invalid negotiation magic %llu (expected %llu)",
-                       Retval->Magic, REPLY_MAGIC);
+        WNBD_LOG_WARN("Received invalid negotiation magic %llu (expected %llu)",
```
Comment (petrutlucian94) [inline driver/nbd_protocol.c:143]: I'm wondering if we should stick to WNBD_LOG_ERROR. It's a fatal error, meaning that the NBD connection cannot be established.

File: driver/scsi_function.c [hunk 8]
```diff
@@ -133,7 +133,7 @@ WnbdExecuteScsiFunction(PVOID DeviceExtension,
     if (SrbGetCdb(Srb)) {
         BYTE CdbValue = SrbGetCdb(Srb)->AsByte[0];
 
-        WNBD_LOG_INFO(": Received %#02x command. SRB = 0x%p. CDB = 0x%x. PathId: %d TargetId: %d LUN: %d",
+        WNBD_LOG_INFO("Received %#02x command. SRB = 0x%p. CDB = 0x%x. PathId: %d TargetId: %d LUN: %d",
```
Comment (petrutlucian94) [inline driver/scsi_function.c:136]: Let's use a debug message instead.

File: driver/scsi_operation.c [hunk 9]
```diff
@@ -629,13 +629,13 @@ WnbdPendOperation(_In_ PWNBD_EXTENSION DeviceExtension,
         }
         break;
     default:
-        WNBD_LOG_ERROR("Unknown Pending SCSI Operation received");
+        WNBD_LOG_INFO("Unknown Pending SCSI Operation received");
```
Comment (petrutlucian94) [inline driver/scsi_operation.c:632]: Let's update the message to WNBD_LOG_INFO("Unknown SCSI operation: %#x", Cdb->AsByte[0])

File: driver/userspace.c [hunk 10]
```diff
@@ -374,21 +374,21 @@ WnbdCreateConnection(PWNBD_EXTENSION DeviceExtension,
     if (!Device->Properties.BlockSize || !Device->Properties.BlockCount ||
         Device->Properties.BlockCount > ULLONG_MAX / Device->Properties.BlockSize)
     {
-        WNBD_LOG_ERROR("Invalid block size or block count. "
-                       "Block size: %d. Block count: %lld.",
-                       Device->Properties.BlockSize,
-                       Device->Properties.BlockCount);
+        WNBD_LOG_WARN("Invalid block size or block count. "
```
Comment (petrutlucian94) [inline driver/userspace.c:377]: Let's stick to an error, we failed to create the connection because of that.

Probably a good rule of thumb: if we manage to resume the connection after the error we can use a warning, if not, we should probably use error level.

File: driver/userspace.c [hunk 11]
```diff
@@ -716,20 +716,20 @@ WnbdParseUserIOCTL(PWNBD_EXTENSION DeviceExtension,
...
         if (!strlen((PSTR) &StatsCmd->InstanceName)) {
-            WNBD_LOG_ERROR("WNBD_STATS: Invalid instance name");
+            WNBD_LOG_WARN("WNBD_STATS: Invalid instance name");
             Status = STATUS_INVALID_PARAMETER;
             break;
         }
 
         if (!Irp->AssociatedIrp.SystemBuffer ||
-                CHECK_O_LOCATION(IoLocation, WNBD_DRV_STATS)) {
+            CHECK_O_LOCATION(IoLocation, WNBD_DRV_STATS)) {
```
Comment (petrutlucian94) [inline driver/userspace.c:732]: Unnecessary change, now it's more difficult to see where the if statement ends.

File: driver/userspace.c [hunk 12]
```diff
@@ -650,7 +650,7 @@ WnbdParseUserIOCTL(PWNBD_EXTENSION DeviceExtension,
         if (!Irp->AssociatedIrp.SystemBuffer ||
             CHECK_O_LOCATION_SZ(IoLocation, RequiredBuffSize))
         {
-            WNBD_LOG_ERROR("IOCTL_WNBD_LIST: Bad output buffer");
+            WNBD_LOG_WARN("IOCTL_WNBD_LIST: Bad output buffer");
```
Comment (petrutlucian94) [inline driver/userspace.c:653]: Let's use a debug message here. Each list command will get a warning since we're adjusting the buffer size.

File: driver/userspace.c [hunk 13]
```diff
@@ -908,7 +908,7 @@ WnbdParseUserIOCTL(PWNBD_EXTENSION DeviceExtension,
             (PWNBD_IOCTL_LIST_DRV_OPT_COMMAND) Irp->AssociatedIrp.SystemBuffer;
 
         if (!ListOptCmd || CHECK_I_LOCATION(IoLocation, IOCTL_WNBD_LIST_DRV_OPT)) {
-            WNBD_LOG_ERROR("IOCTL_WNBD_RESET_DRV_OPT: Bad input or output buffer");
+            WNBD_LOG_WARN("IOCTL_WNBD_RESET_DRV_OPT: Bad input or output buffer");
```
Comment (petrutlucian94) [inline driver/userspace.c:911]: Let's use a debug message here. Each list command will get a warning since we're adjusting the buffer size.

File: driver/wnbd_dispatch.c [hunk 14]
```diff
@@ -297,15 +297,15 @@ NTSTATUS WnbdHandleResponse(
...
+                WNBD_LOG_WARN("Could not get SRB %p 0x%llx data buffer. Error: %lu.",
+                              Element->Srb, Element->Tag, StorResult);
                 Element->Srb->SrbStatus = SRB_STATUS_INTERNAL_ERROR;
                 Status = STATUS_INTERNAL_ERROR;
                 goto Exit;
             }
         }
     } else {
-        WNBD_LOG_WARN("Received reply header for aborted request: %p 0x%llx.",
+        WNBD_LOG_INFO("Received reply header for aborted request: %p 0x%llx.",
```
Comment (petrutlucian94) [inline driver/wnbd_dispatch.c:308]: Ditto, let's use a debug message.

**Review by petrutlucian94 — CHANGES_REQUESTED** [review]

File: driver/driver.c (hunk 3 above)
Comment (aserdean) [reply driver/driver.c:106]: It's easier to see the changes in my opinion, but sure I will squash them in one commit.

File: driver/driver.c (hunk 2 above)
Comment (aserdean) [reply driver/driver.c:32]: It will help in the event of stack corruption (very small chance for it to happen, please let me know if you want to keep them).

File: driver/driver.c (hunk 1 above)
Comment (aserdean) [reply driver/driver.c:92]: Sure, I will change the rest of the occurrences.

File: driver/userspace.c (hunk 11 above)
Comment (aserdean) [reply driver/userspace.c:732]: I will revert the change, although both seem terrible. I'm wondering if we should move the logical operator on a new line, that will help with the visibility.

File: driver/nbd_dispatch.c [hunk 15]
```diff
@@ -184,12 +177,12 @@ NbdDeviceRequestThread(_In_ PVOID Context)
     }
 
     PsTerminateSystemThread(STATUS_SUCCESS);
+    return;
```
Comment (petrutlucian94) [inline driver/nbd_dispatch.c:180]: Unrelated change, probably mistake.

File: driver/nbd_dispatch.c [hunk 16]
```diff
@@ -201,12 +194,12 @@ NbdDeviceReplyThread(_In_ PVOID Context)
...
```
Comment (petrutlucian94) [inline driver/nbd_dispatch.c:197]: ditto

File: driver/nbd_protocol.c [hunk 17]
```diff
@@ -392,7 +373,7 @@ NbdOpenAndConnect(PCHAR HostName,
     if (Ai) {
         FreeAddrInfo(Ai);
     }
-    WNBD_LOG_LOUD(": Exit");
+    WNBD_LOG_LOUD("Exit");
```
Comment (petrutlucian94) [inline driver/nbd_protocol.c:376]: Leftover

File: driver/nbd_protocol.c [hunk 18]
```diff
@@ -433,7 +413,7 @@ NbdRequest(
     }
 Exit:
     *IoStatus = Status;
-    WNBD_LOG_LOUD(": Exit");
+    WNBD_LOG_LOUD("Exit");
```
Comment (petrutlucian94) [inline driver/nbd_protocol.c:416]: Ditto.

File: driver/userspace.c (hunk 11 above)
Comment (petrutlucian94) [reply driver/userspace.c]: Yep, that would work as well.

File: driver/nbd_dispatch.c (hunk 4 above)
Comment (petrutlucian94) [reply driver/nbd_dispatch.c]: Ughh, my bad. We must use WNBD_LOG_ERROR for nbd_dispatch since that will close the NBD connection. It's a non-fatal error for wnbd_dispatch, where I see that you're already using WNBD_LOG_TRACE (might be worth using WNBD_LOG_DEBUG for consistency).

**Review by petrutlucian94 — APPROVED** [review]
Looks good, let's wait for the tests to complete and then we can merge it. Thanks for addressing the comments!
